OutOfSchool/OutOfSchool.BusinessLogic/Services/CompetitiveEventService.cs

```diff
@@ -24,6 +26,14 @@ public class CompetitiveEventService : ICompetitiveEventService
     private readonly ICurrentUserService currentUserService;
     private readonly IContactsService<CompetitiveEvent, IHasContactsDto<CompetitiveEvent>> contactsService;
 
+    private readonly Func<IQueryable<CompetitiveEvent>, IQueryable<CompetitiveEvent>> includeFunc =
+    query => query
+        .Include(e => e.InstitutionHierarchy)
+        .Include(e => e.CompetitiveEventDescriptionItems)
+        .Include(e => e.Coverage)
+        .Include(e => e.Contacts)
+            .ThenInclude(c => c.Address)
+                .ThenInclude(a => a.CATOTTG);
```

Comment:
to fully populate contacts address without exceptions you will have to do something like this:
```csharp
.ThenInclude(e => e.Contacts).ThenInclude(c => c.Address).ThenInclude(a => a.CATOTTG)
                .ThenInclude(c => c.Parent).ThenInclude(c => c.Parent).ThenInclude(c => c.Parent).ThenInclude(c => c.Parent)
```

so we need to have catottg with 4 levels of parents.